kalite/coachreports/static/js/coachreports/coach_reports/views.js

```diff
@@ -0,0 +1,343 @@
+
+        this.render();
+    },
+
+    render: function() {
+        this.$el.html(this.template());
+        this.$('#group-select-container').append(this.group_select_view.$el);
+        this.$('#facility-select-container').append(this.facility_select_view.$el);
+        this.tabular_view = new TabularReportView({model: this.model});
+        this.$("#student_report_container").append(this.tabular_view.$el);
+    }
+});
+
+var DetailPanelInlineRowView = BaseView.extend({
+
+    tagName: 'tr',
+
+    className: 'details-row',
+
+    initialize: function(options) {
+        this.contents_length = options.contents_length;
+        this.content_item = options.content_item;
+        this.render();
+    },
+
+    render: function() {
+        this.detail_view = new detailsPanelView({
+            tagName: 'td',
+            model: this.model,
+            content_item: this.content_item,
+            attributes: {colspan: this.contents_length + 1}
+        });
+        this.$el.append(this.detail_view.el);
+    }
+})
+        
+var detailsPanelView = BaseView.extend({
+    
+    //Number of items to show from the collection
+    limit: 4,
+    
+    template: HB.template("coach_nav/detailspanel"),
+    
+    initialize: function (options) {
+        _.bindAll(this);
+        this.content_item = options.content_item;
+        if (this.content_item.get("kind") === "Exercise") {
+            this.collection = new window.AttemptLogCollection([], {
+                user: this.model.get("user"),
+                limit: this.limit,
+                exercise_id: this.model.get("exercise_id")
+            });
+            this.collection.fetch();
+        }
+        this.listenToOnce(this.collection, "sync", this.render);
+        this.render();
+    },
+    
+    render: function() {
+        var item_count = 0;
+        if (this.collection.meta) {
+            item_count = this.collection.meta.total_count;
+        }
+        this.pages = [];
+        if (item_count/this.limit > 1) {
+            for (var i=1; i < item_count/this.limit + 1; i++) {
+                this.pages.push(i);
+            }
+        }
+        this.$el.html(this.template({
+            model: this.model.attributes,
+            itemdata: this.content_item.attributes,
+            pages: this.pages,
+            collection: this.collection.to_objects()
+        }));
+        this.bodyView = new detailsPanelBodyView ({
+            collection: this.collection,
+            el: this.$(".body")
+        });
+    }
+});
+
+
+var detailsPanelBodyView = BaseView.extend({
+    
+    template: HB.template("coach_nav/detailspanelbody"),
+    
+    initialize: function (options) {
+        _.bindAll(this);
+        this.render();
+    },
+    
+    render: function() {
+        this.$el.html(this.template({
+            collection: this.collection.to_objects()
+        }));
+    }
+});
+
+var TabularReportView = BaseView.extend({
```

Comment:
Lots of different views here, so for clarity I think it would be useful to explicitly spell out the hierarchy of views in this document, like
```yml
CoachReportView:
  - FacilitySelectView
  - GroupSelectView
  - TabularReportView:
    - [subviews go here]
```